### PR TITLE
Mark invariant normalizer in Gemma as non-persistent

### PR DESCRIPTION
**Pull request by yhtang**
Prevents the Gemma embedding normalizer, which is a model invariant, from being scrambled by the dummy weight loader.

This is a quick fix for the problem when a vLLM Gemma model keeps producing gibberish in an RLHF setup (similar to the [RLHF example](https://github.com/vllm-project/vllm/blob/main/examples/offline_inference/rlhf.py)) if it was initially loaded with dummy weights, _even after_ we send in the real weights. It is due to the dummy weight loader [traversing `state_dict` rather than `named_parameters`](https://github.com/vllm-project/vllm/blob/d4629dc43f92c189e90a3e2f2f8a52648aed4d9d/vllm/model_executor/model_loader/weight_utils.py#L689) and inadvertently randomizing the normalizer, which is a model invariant rather than a trainable weight.

A quick scan of the model registry indicates that other models register similar invariant buffers with `persistent=False`, thus not triggering this issue.

A long-term solution would be to provide an API for marking a buffer as invariant, e.g.
```
def register_invariant(self, name, tensor):
    tensor._no_dummy_init = True
    self.register_buffer(name, tensor)
```
and have `initialize_dummy_weights` respect that flag.

### CODE REVIEW

File: tests/models/language/generation/test_gemma.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            model,
+            load_format="dummy",
+    ) as llm:
+        normalizers = llm.collective_rpc(
+            lambda self: self.worker.model_runner.model.model.normalizer.cpu().item()
+        )
+        assert np.allclose(
+            normalizers,
+            llm.llm_engine.model_config.hf_config.hidden_size**0.5,
+            rtol=1e-3
```
[the remark this answers is not in the document]
Comment (yhtang) [reply tests/models/language/generation/test_gemma.py]: The assertion will fail with rtol=1e-4. This is due to the tensor using lower-precision floats such as bf16.

**Review by DarkLight1337 — APPROVED** [review]
Thanks for fixing, please resolve the pre-commit errors

**Comment by yhtang** [comment]
> Thanks for fixing, please resolve the pre-commit errors

Thanks for reviewing! I've fixed the format errors, but it's now complaining again about a file that is not in this PR.

**Review by DarkLight1337 — APPROVED** [review]
